### ISSUE TITLE
Enunciado apartado de DESCONEXIÓN

### ISSUE THREAD

**Issue by ricardoAguirreSanchez** [sisoputnfrba/foro#1067]
Una consulta sobre el enunciado:

Desconexión
Durante la ejecución del Sistema, existe la posibilidad de que una o más Instancias dejen de estar disponibles para el Coordinador de Re Distinto. Ante éstas situaciones, el Coordinador deberá ajustar su distribución de forma acorde a la cantidad de Instancias disponibles. No se deberá eliminar una clave si la instancia de desconectó; solo se elimina cuando un ESI intenta acceder a ella; de tal forma, si la instancia se reincorpora previo al uso de la clave; el sistema sigue funcionando.

Cuando dice "no se deberá eliminar la clave" se refiere a no eliminar de donde? La clave en teoría está guardado dentro de la instancia y posiblemente en alguna tabla dentro del coordinador, osea sería no eliminar de la tabla del coordinador? 

Saludos

**Comment by iago64** [comment]
Buenas buenas!

Perdón la demora, la clave no se debería eliminar de la lista de claves que tiene el coordinador, ya que a futuro la instancia puede reconectarse.

Saludos.-

**Comment by ricardoAguirreSanchez** [comment]
Bien, 
Entonces por un lado en coordinador tendrá una lista de claves que "existen" en el sistema, estás claves se van creando confirme se hacen las operaciones GET ?

**Comment by afilgueira** [comment]
Buenas, el coordinador tiene una lista de claves y sabe en que instancia estan (si ya estan en una). Desde el coordinador no necesitas saber nada del ESI, simplemente respondes cosas. El que debe saber los bloqueos por clave y ESI es el planificador.

Saludos!

**Comment by ricardoAguirreSanchez** [comment]
Y si la operación es un SET? En el enunciado dice que no se puede hacer a menos que el esi que lo pidió tenga previamente la clave tomada (GET), pero la operación SET no va al planificador "normalmente", una alternativa a esto es (aunque no se indique en el flujo normal) mandar un mensaje al planificador y pedirle que me indique si éste esi puede o no hacer el SET?

**Comment by afilgueira** [comment]
Buenas!

Por qué el SET no podría ir al planificador "normalmente"? Hasta ahora no era necesario, pero entiendo que no debería traer problemas tirarle un ping (y esto evita todos las otras cosas que venían consultando).

Si ves algo que no vimos avisanos!

**Comment by ricardoAguirreSanchez** [comment]
Claro, cuando dije que no va normalmente me refería a que no era parte del flujo de una sentencia SET, entiendo que el SET le llega al coordinador y el determina a que instancia mandarlo.

Ahora, sí para evitarme el trabajo de tener que desde el coordinador soportar una estructura con esis y claves puedo mandar un mensaje al planificador y que esté me diga si se puede hacer el set estaría solucionado mí problema.
